### ISSUE TITLE
custom field of type URL: Scheme mailto not working

### ISSUE THREAD

**Issue by idefax** [joomla/joomla-cms#37029]
### Steps to reproduce the issue
Go to articles. There create a custom field of type "URL". For scheme select "mailto" and save.

### Expected result
If adding/editing an article and enter a valid email address into the custom URL field, in the frontend a proper email link should be generated.

### Actual result
A https link including email address is generated instead, such as: https://www.mydomain.dd/firstname-lastname@mydomain.dd

### System information (as much as possible)
Joomla 4.0.6, PHP 7.4

### Additional comments
This issue also exists in J3.

**Comment by esedic** [comment]
I have tested this on Joomla 4.0.6. and I can't confirm this. You need to add URL schema (mailto:) to the field and email link is displayed correctly.

Although there are other issues, like mailto: schema displayed in email link.

**Comment by idefax** [comment]
Agree: If I add schema manually in front of email address, then email addresss (href) is build correctly but "mailto:" is visible in Frontend.

Since I select "mailto" from schema dropdown already, I would expect not to have to add it manually as well.  And "mailto:" beeing visible in frontend is not good.

**Comment by brianteeman** [comment]
@idefax Its a multi-select box used to validate the type of link

**Comment by idefax** [comment]
@brianteeman But in the article, only one type of data will be entered into the field: either email, http URL, file etc. Why multi-select?
Besides this, adding schema "mailto:" as well in front of the email address might be no problem for experienced users, but it's not for not so experienced users. They would expect to enter email address only and link will be build corrrectly. My toughts.

And as mentioned in comments earlier: "mailto:" is visible in frontend.

**Comment by Vanshitasingh** [comment]
Hey @idefax , could you elaborate a further more on how to reproduce this issue? I don't know if I did it correctly but I created a Url field and linked it with a valid email and it worked perfectly fine for me.

**Comment by Vanshitasingh** [comment]
<img width="919" alt="image" src="https://user-images.githubusercontent.com/64356829/160563117-b05a48cb-b0f6-462a-bb92-8f0441f50bf6.png">
As you can see, I've linked the text using mailto: and it opened the mail as expected

**Comment by idefax** [comment]
@Vanshitasingh 

**Step 1)**
I open Articles > Fields. There I add a custom field of type "URL". In its settings I set parameter for schema to "Mailto":

![Screenshot_1](https://user-images.githubusercontent.com/22318189/160566963-3cfcf7e0-9476-4df2-a5be-57c2640bc15b.jpg)


**Step 2)**
After I edit an article and add email address into the new URL field. When saving, I get a warning message about missing schema, even if I have set it previously in step 1):

![Screenshot_2](https://user-images.githubusercontent.com/22318189/160567513-c7fd8de8-b9c8-4f07-a3f5-6721cb294d41.jpg)

**Step 3)**
I add "mailto:" into email field:

![Screenshot_3](https://user-images.githubusercontent.com/22318189/160567629-881602eb-0d60-4291-a946-d2abab362a7e.jpg)


**Result:** 
Email link works in frontend, but schema is visible to user:

![Screenshot_4](https://user-images.githubusercontent.com/22318189/160567871-1f3de185-7d47-4bf6-8eb8-37ec2be082ba.jpg)


**My thoughts:**
- why do I need to add schema into URL field even if I have set parameter for schema properly in the settings of custom field?
- "mailto:" should not be visible to user in frontend
